### DISCUSSION TITLE
How to use local hosted model with vllm as a custom llm-as-a-judge?

### DISCUSSION THREAD

**Discussion by sir3mat** [mlflow/mlflow#12763] [Q&A] [answered]
How to use local hosted model with vllm as a custom llm-as-a-judge?

It seems that for evaluation using llm as a judge the only possible model is openai. Any tips?

**Comment by sir3mat** [comment]
maybe it would be helpful to change this url of openai sdk (https://github.com/mlflow/mlflow/blob/0b1d995d66a678153e01ed3040f3f4dfc16a0d6b/mlflow/metrics/genai/model_utils.py#L98)

**Reply by B-Step62** [reply, marked as answer]
To use a locally hosted model with vllm as a custom LLM-as-judge, you would need to set up your local MLflow Deployments Server. The default judge model for LLM-as-judge metrics is OpenAI's GPT-4, you can override this by specifying your local model endpoint in the metric definition.

1. **Set Up Your Local MLflow Deployments Server** - Assuming your model conforms to the OpenAI API spec, you can create an MLflow Deployments Server with provider `openai` and set `openai_api_base` key to your local vllm endpoint. Please refer to [MLflow Deployments Server documentation](https://mlflow.org/docs/latest/llms/deployments/index.html) and [OpenAI specific configuration](https://mlflow.org/docs/latest/llms/deployments/index.html#id1) for more details.
2. **Set the Deployment Target** - Use the `mlflow.deployments.set_deployments_target()` function to point to your local server. For example: 
```python
mlflow.deployments import set_deployments_target
set_deployments_target("http://localhost:5000/")
```
3. **Define Your Custom Metric** - When defining your custom metric, specify your local model endpoint in the `model` argument. For example:
```
my_custom_metric = mlflow.metrics.genai.make_genai_metric(
    name="my_custom_metric",
    definition="This metric evaluates the quality of responses.",
    model="endpoints:/my-local-vllm-endpoint"  # Specify the local MLFlow Deployments Server endpoint
    grading_prompt="Rate the professionalism of the response.",
    examples=[...],  # Provide examples for the LLM judge
)
```
For more details, please refer to [Selecting the LLM-sa-judge Model](https://mlflow.org/docs/latest/llms/llm-evaluate/index.html#selecting-the-llm-as-judge-model)

**Reply by sir3mat** [reply]
If I deployed MLFlow using a Helm chart, can I also deploy the MLFlow model deployment server?
@B-Step62

**Reply by sir3mat** [reply]
Ok I deployed correctly my pod o k8s and metrics are now running correctly! Thanks!

How can I connect the UI with the deployments server? There are any env variables to set in order to have the playground available?

**Comment by josiahrohrer** [comment]
I tried the same thing with the new make_judge function which does not seem to work for me, is it intended to work the same?
